src/DotNetBoilerplate.Core/Catalogs/Catalog.cs

```diff
@@ -0,0 +1,65 @@
+﻿using DotNetBoilerplate.Core.Books;
+using DotNetBoilerplate.Core.Catalogs.Exceptions;
+using DotNetBoilerplate.Core.Users;
+
+namespace DotNetBoilerplate.Core.Catalogs;
+
+public sealed class Catalog
+{
+    private Catalog()
+    { }
+    public Guid Id { get; private set; }
+    public string Name { get; private set; }
+    public string Genre { get; private set; }
+    public string Description { get; private set; }
+    public List<Book> Books { get; set; }
+    public Guid BookStoreId { get; private set; }
+    public UserId CreatedBy { get; private set; }
+    public DateTime UpdatedAt { get; private set; }
+
+    public static Catalog Create(
+        string name,
+        string genre,
+        string description,
+        Guid bookStoreId,
+        Guid createdBy,
+        DateTime updatedAt,
+        bool userCanNotCreateCatalog
+        )
+    {
+        if (userCanNotCreateCatalog)
+            throw new UserCanNotCreateCatalogException();
+
+        return new Catalog
+        {
+            Id = Guid.NewGuid(),
+            Name = name,
+            Genre = genre,
+            Description = description,
+            Books = new List<Book>(),
+            BookStoreId = bookStoreId,
+            CreatedBy = createdBy,
+            UpdatedAt = updatedAt
+        };
+    }
+    public void Update(
+        string name,
+        string genre,
+        string description,
+        Guid updater,
+        DateTime updatedAt,
+        bool userCanNotUpdateCatalog
+
+    )
+    {
+        if (updater == Guid.Empty)
+            throw new UserCanNotUpdateCatalogException();
+        if (userCanNotUpdateCatalog)
```

Comment:
userCanNotUpdateCatalog - ten parametr nic nie mówi co to sprawdza - trzeba lepiej nazwać